File: backend/src/main/java/org/providcy/acceptance/pagination/Pager.java
package org.providcy.acceptance.pagination;

import lombok.Data;
import org.springframework.data.domain.Page;

@Data
public class Pager {

    private int currentPage;
    private int end;
    private boolean hasNextPage;
    private boolean hasPreviousPage;
    private int maxPerPage;
    private int nbPages;
    private long nbResult;
    private long start;

    public Pager() {
    }

    public <T> Pager(Page<T> page) {
        this.currentPage = page.getNumber() + 1;
        this.end = page.getNumberOfElements();
        this.hasNextPage = page.hasNext();
        this.hasPreviousPage = page.hasPrevious();
        this.maxPerPage = page.getSize();
        this.nbPages = page.getTotalPages();
        this.nbResult = page.getTotalElements();
        this.start = !page.isEmpty() ? page.getPageable().getOffset() : 0;
    }
}

File: frontend/projects/tools/src/lib/module/common/model/pager.ts
export interface IPager {
    currentPage: number;
    end: number;
    hasNextPage: boolean;
    hasPreviousPage: boolean;
    maxPerPage: number;
    nbPages: number;
    nbResult: number;
    start: number;
}

export class Pager implements IPager {
    constructor(
        public currentPage: number = 1,
        public end: number = 1,
        public hasNextPage: boolean = false,
        public hasPreviousPage: boolean = false,
        public maxPerPage: number = 10,
        public nbPages: number = 1,
        public nbResult: number = 0,
        public start: number = 1
    ) {}
}
